packages/cody/src/lib/hooks/on-aggregate.ts

```diff
@@ -91,6 +91,7 @@ export const onAggregate: CodyHook<Context> = async (aggregate: Node, ctx: Conte
 
     withErrorCheck(register, [aggregate, ctx, tree]);
     withErrorCheck(registerCommandHandler, [service, aggregate, ctx, tree]);
+    withErrorCheck(registerAggregateRepository, [service, aggregate, ctx, tree]);
```

Comment:
All functions in the Cody package return a `CodyResponse` of type `CodyResponseType.Error` whenever they hit a problem. This ensures that all code generation related errors are returned back to prooph board and are correctly shown as errors in the Cody Console.

The downside of this approach is that the code is a little bit bloated because you always have to check if a function returned a Cody error instead of the expected return type.

The `withErrorCheck` function is a little helper to avoid that bloated code in the Cody hooks. If it detects a Cody Error it throws a specific exception that is caught at the end of the hook and returned back into a CodyResponse for prooph board.